agent/ansible/pbench/agent/Makefile

```diff
@@ -0,0 +1,21 @@
+ROLES = $(wildcard  roles/*)
+NROLES = $(shell echo ${ROLES} | wc -w)
+README = $(wildcard roles/*/README.md)
+NREADME= $(shell echo ${README} | wc -w)
+
+
+all:	lint check-readme
+
+lint:
+	ansible-lint -c ansible-lint.yml
+	yaml-lint $$(find . -type f -name '*.yml')
+
+check-readme:
+	@if [ ${NROLES} != ${NREADME} ] ;then\
+		echo "Mismatch between roles and README.md files";\
+		echo "roles: ${ROLES}";\
+	     	echo "role README files: ${README}";\
+		exit 1\
+	;fi
```

Comment:
I figured that I could stand the pain of comparing the two lists that are printed out and figuring out the differences. The problem with the loop is that it will exit on the first error, so you'll need to `make` again to catch the second. But as I said, I should write a little checker for this.

Reply:
Ah!  I didn't notice that it was printing the contents of the lists (I was mistakenly thinking it was printing the counts).

Now that I've had supper, let me see if I can puzzle out a properly `make`-ish way to handle this...it should be pretty simple.

Reply:
Here's one possible approach.

Replace the variable definitions at the beginning of the `Makefile` with these:
```
ROLES = $(wildcard  roles/*)
READMES = README.md plugins/README.md $(patsubst %,%/README.md,$(ROLES))
```
Replace the `check-readme` target with this:
``` 
check-readme: $(READMES)
	@for i in $?; do \
	  if [[ ! -f $$i ]]; then exit 1; fi; \
	done
```
After checking the prerequisites, it simply decides whether to exit with an error or not.  The real work is done in the prerequisites:
```
$(READMES):
	@echo $@ is missing.
```
So, the `check-readme` target will print out a list of missing `README.md` files and then exit with a failure status if any are missing; if they are all there, then nothing is printed and no error is issued.  The magic is in the definition of `READMES` which builds a list of all the `README.md` files which are supposed to exist, based on the `roles` subdirectories, with the addition of the ones for the plug-in directory and the top level.

Reply:
FWIW, it is easy to get it to print an error and fail on the first missing file.  Likewise, it's easy to get it to print all of the missing files.  The challenge was to get it to print all the missing files _and_ fail...that required the loop in `check-readme`.

Reply:
Actually, if we're gonna have a loop in `check-readme`, then we can just inline your script there:
```
check-readme: 
	@for i in $(READMES); do \
	  if [[ ! -f $${i} ]]; then \
	    echo "$${PWD}/$${i} is missing or not a regular file"; \
	    status=1; \
	  fi; \
	done; \
	exit $${status}
```
We remove the prerequisite list and iterate over it directly; we don't need the extra target to do the printing, since we're visiting all the files anyway; and, because this is all one shell command, we can carry the `status` value through to the `exit`.

Reply:
I'll leave things as they are for now at least. If/when we migrate the ansible stuff to its own repo, we can revisit it.